lib/compiler.js

```diff
@@ -12,7 +12,13 @@ const AV = require("./assignment-visitor.js");
 const assignmentVisitor = new AV;
 
 const shebangRegExp = /^#!.*/;
-const importExportRegExp = /\b(?:im|ex)port\b/;
+
+// Matches any import or export identifier as long as it's not preceded by
+// a `.` character (to avoid matching module.export, for example). Because
+// Reify replaces all import and export declarations with module.import
+// and module.export calls, this logic should prevent the compiler from
+// ever having to recompile code it has already compiled.
+const importExportRegExp = /(?:^|[^.])\b(?:im|ex)port\b/;
```

Comment:
There's also a balance between a quick check and the cost of a complex regexp so I'd side on simplicity over complex.... there are regexp helpers that help point out the complexity of a regexp.